wake-lock/wakelock-query.https.any.js

```diff
@@ -0,0 +1,144 @@
+    () => {},
+    self
+  ];
+  const lock = new WakeLock("screen");
+
+  invalidTypes.map(invalidType => {
+    assert_throws(new TypeError(), () => lock.query({ type: invalidType }));
+  });
+}, "'TypeError' is thrown when filter WakeLock with an invalid type");
+
+test(() => {
+  const lock1 = new WakeLock("screen");
+  const lock2 = new WakeLock("screen");
+  const lock3 = new WakeLock("system");
+  const lock4 = new WakeLock("system");
+
+  const lockList = lock1.query();
+  assert_equals(lockList.size, 4, "Size of WakeLock instance list");
+  for (let lock of lockList) {
+    assert_true(lock instanceof WakeLock, "query() must have returned WakeLock instance");
+  }
+}, "Filter all constructed WakeLock instances");
+
+test(() => {
+  const lock1 = new WakeLock("screen");
+  const lock2 = new WakeLock("screen");
+  const lock3 = new WakeLock("system");
+  const lock4 = new WakeLock("system");
+
+  // query "screen" type of WakeLock
+  const lockList1 = lock1.query({ type: "screen" });
+  assert_equals(lockList1.size, 2, "Size of WakeLock instance list with 'screen' type");
+  for (let lock of lockList1) {
+    assert_true(lock instanceof WakeLock, "query() must have returned WakeLock instance");
+    assert_equals(lock.type, "screen", "WakeLock type");
+  }
+
+  // query "system" type of WakeLock
+  const lockList2 = lock1.query({ type: "system" });
+  assert_equals(lockList2.size, 2, "Size of WakeLock instance list with 'system' type");
+  for (let lock of lockList2) {
+    assert_true(lock instanceof WakeLock, "query() must have returned WakeLock instance");
+    assert_equals(lock.type, "system", "WakeLock type");
+  }
+}, "Filter constructed WakeLock instances with 'type' option");
+
+promise_test(async t => {
+  const lock1 = new WakeLock("screen");
+  const lock2 = new WakeLock("screen");
+
+  const controller = new AbortController();
+  await lock1.request({ signal: controller.signal });
+  assert_true(lock1.active, "lock1 must have been activated");
+  assert_true(lock2.active, "lock2 must have been activated");
+
+  // query activated WakeLock
+  const lockList1 = lock1.query({ active: true });
+  assert_equals(lockList1.size, 2, "Size of WakeLock instance list with activated state");
+  for (let lock of lockList1) {
+    assert_true(lock instanceof WakeLock, "query() must have returned WakeLock instance");
+    assert_equals(lock.type, "screen", "WakeLock type");
+  }
+
+  // query inactivated WakeLock
+  const lockList2 = lock1.query({ active: false });
+  assert_equals(lockList2.size, 0, "Size of WakeLock instance list with activated state");
+
+  controller.abort();
+}, "Filter constructed WakeLock instances with 'active' option");
+
+promise_test(async t => {
+  const lock1 = new WakeLock("screen");
+  const lock2 = new WakeLock("screen");
+  const lock3 = new WakeLock("system");
+  const lock4 = new WakeLock("system");
+
+  const controller = new AbortController();
+  await lock1.request({ signal: controller.signal });
```

Comment:
Notice that we agreed on removing abortcontroller and there is a WIP PR for it

Reply:
Yeah, I knew the change, and will replace all `AbortController` with `release()` once your PR is landed.